sql/stats: TestAddMisestimateRandomized failed

sql/stats.TestAddMisestimateRandomized [failed](https://mesolite.cluster.engflow.com/invocations/default/c2cffbce-f3d1-4a6f-b0eb-64683354463a?testReportRun=1&testReportShard=7&testReportAttempt=1#targets-Ly9wa2cvc3FsL3N0YXRzOnN0YXRzX3Rlc3Q=) on master @ [a41e86dbb2bd18447a4ccf69d6c5f887827cd7d7](https://github.com/cockroachdb/cockroach/commits/a41e86dbb2bd18447a4ccf69d6c5f887827cd7d7):


```
=== RUN   TestAddMisestimateRandomized
    automatic_stats_test.go:1432: 
        	Error Trace:	pkg/sql/stats/automatic_stats_test.go:1432
        	Error:      	Not equal: 
        	            	expected: roachpb.Spans{roachpb.Span{Key:"c", EndKey:"t"}, roachpb.Span{Key:"u", EndKey:"v"}, roachpb.Span{Key:"y", EndKey:"z"}}
        	            	actual  : roachpb.Spans{roachpb.Span{Key:"c", EndKey:"t"}, roachpb.Span{Key:"h", EndKey:"t"}, roachpb.Span{Key:"u", EndKey:"v"}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -4,2 +4,10 @@
        	            	    00000000  63                                                |c|
        	            	+  },
        	            	+  EndKey: (roachpb.Key) (len=1) {
        	            	+   00000000  74                                                |t|
        	            	+  }
        	            	+ },
        	            	+ (roachpb.Span) {
        	            	+  Key: (roachpb.Key) (len=1) {
        	            	+   00000000  68                                                |h|
        	            	   },
        	            	@@ -16,10 +24,2 @@
        	            	   }
        	            	- },
        	            	- (roachpb.Span) {
        	            	-  Key: (roachpb.Key) (len=1) {
        	            	-   00000000  79                                                |y|
        	            	-  },
        	            	-  EndKey: (roachpb.Key) (len=1) {
        	            	-   00000000  7a                                                |z|
        	            	-  }
        	            	  }
        	Test:       	TestAddMisestimateRandomized
--- FAIL: TestAddMisestimateRandomized (0.00s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=7</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAddMisestimateRandomized.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-63078

--- a/pkg/sql/stats/automatic_stats_test.go
+++ b/pkg/sql/stats/automatic_stats_test.go
@@ -9,6 +9,7 @@ import (
 	"context"
 	"fmt"
 	"math"
+	"slices"
 	"sort"
 	"strings"
 	"testing"
@@ -1429,6 +1430,6 @@ func TestAddMisestimateRandomized(t *testing.T) {
 			indexInfo: id,
 			Spans:     new,
 		})
-		require.Equal(t, dedupAndMerge(append(old, new...)), r.misestimateSpans[id])
+		require.Equal(t, dedupAndMerge(append(slices.Clone(old), new...)), r.misestimateSpans[id])
 	}
 }

